src/recap

```diff
@@ -146,492 +134,440 @@ print_usage() {
-                                print_blankline $MYSQL_FILE
-                        fi
-                done
-        fi
+  local LOGFILE=$1
+  # We only support a single DOTMYDOTCNF for MySQL when running Plesk
+  local PLESK_FILE="/etc/psa/.psa.shadow"
+  if [[ -r "${PLESK_FILE}" ]]; then
+    echo "MySQL status" >> "${LOGFILE}"
+    mysqladmin -uadmin -p$(cat "${PLESK_FILE}") status >> "${LOGFILE}"
+    print_blankline "${LOGFILE}"
+  else
+    for cnf in $( sed -e 's/,/ /g' <<<"${DOTMYDOTCNF}" ); do
+      if [[ -r "${cnf}" ]]; then
+        echo "MySQL ${cnf} status" >> "${LOGFILE}"
+        mysqladmin --defaults-extra-file="${cnf}" status >> "${LOGFILE}"
+        print_blankline "${LOGFILE}"
+      else
+        echo "MySQL ${cnf} for status does not exist. Please check DOTMYDOTCNF in /etc/recap" >> "${LOGFILE}"
+        print_blankline "${LOGFILE}"
+      fi
+    done
+  fi
 }
 
 # print the non-truncated innodb status to the mysql file
 print_mysql_innodb_status() {
-        # First, we establish which tmpdir and pid_file are in use, strip any
-        # trailing slash and populate the file with current information.
-        # We throw away the output of the "show engine innodb status" command as
-        # it is likely being truncated, which we attempt to work around:
-
-	echo "MySQL InnoDB status"  >> $MYSQL_FILE
-	unset MYVALS PID_FILE TMPDIR
-	MYVALS=($(IFS=$'\t' mysql -Bse "SHOW VARIABLES LIKE 'pid_file'; \
-                            SHOW VARIABLES LIKE 'tmpdir'; \
-                            SHOW ENGINE INNODB STATUS;" 2>/dev/null \
-                           | head -n 2))
-        PID_FILE=${MYVALS[1]%/}
-        TMPDIR=${MYVALS[3]%/}
-
-        # Next, we grab a list of descriptors in the tmpdir:
-
-        for name in "/proc/"$(cat $PID_FILE)"/fd/"*;
-
-        # We know that InnoDB's temporary files are always in the mysql tmpdir
-        # and start with the string 'ib' followed by a randomly generated series
-        # of characters.  Now we can compare the canonicalized path of fd to the
-        # tmpdir/ib* pattern:
-
-        do [[ $(readlink -f "$name") == "$TMPDIR"/ib* ]] &&
-
-        # If any files match that pattern, we see if the first line in the file
-        # starts with "===", which is the case in the particular InnoDB file we
-        # care about:
-
-        head -c8 "$name" | grep -q '^====' &&
-
-        # Finally, print out the information and append it to the mysql file:
-        cat "$name"  >> $MYSQL_FILE
-	done
+  local LOGFILE=$1
+  # First, we establish which tmpdir and pid_file are in use, strip any
+  # trailing slash and populate the file with current information.
+  # We throw away the output of the "show engine innodb status" command as
+  # it is likely being truncated, which we attempt to work around:
+
+  echo "MySQL InnoDB status"  >> "${LOGFILE}"
+  unset MYVALS PID_FILE TMPDIR
+  MYVALS=( $( IFS=$'\t' mysql -Bse "SHOW VARIABLES LIKE 'pid_file'; \
```

Comment:
There are three functions for mysql interactions, only two of them loop through multiple mysql configuration: `print_mysql`, `print_mysql_procs`, but not for `print_mysql_innodb_status` as you pointed out.  All these functions are being called from: `run_mysql_report` this makes me think if we should instead of looping inside each `print` function or from the `run` function, need to analyze this better and take a decision, this will come in a separate commit.